src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java

```diff
@@ -74,23 +94,55 @@ public void resetState(PhpSnippetSet<?> phpSnippetSet, PhpContextCommon phpCommo
   // ===============
 
   @Override
+  /**
+   * Returns the method names used in the PHP client code.
+   */
   public String getMethodName(Method method) {
     String name = super.getMethodName(method);
-    List<String> resources = getApiaryConfig().getResources(method.getName());
-    if (resources != null && resources.size() > 0) {
-      name = name + lowerCamelToUpperCamel(resources.get(resources.size() -1));
+    if (RENAMED_METHODS.contains(name)) {
+      List<String> resources = getApiaryConfig().getResources(method.getName());
+      for (int i = 0; i < resources.size(); i++) {
+        name = name + lowerCamelToUpperCamel(resources.get(i));
```

Comment:
Consider using `for (String resource : getResources())`

Consider using StringBuilder.